src/Microsoft.Diagnostics.Monitoring.EventPipe/DistributedTraces/DistributedTracesPipeline.cs

```diff
@@ -0,0 +1,150 @@
+                                cancellationTokenSource.Cancel();
+                            }
+                        }
+                        catch (Exception)
+                        {
+                        }
+                    };
+
+                    using EventTaskSource<Action> sourceCompletedTaskSource = new(
+                        taskComplete => taskComplete,
+                        handler => eventSource.Completed += handler,
+                        handler => eventSource.Completed -= handler,
+                        token);
+
+                    await sourceCompletedTaskSource.Task.ConfigureAwait(false);
+                });
+
+            try
+            {
+                await processor.Process(Client, TimeSpan.FromSeconds(10), resumeRuntime: false, token: cancellationTokenSource.Token).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
+            }
+
+            await processor.DisposeAsync().ConfigureAwait(false);
+
+            if (majorVersion < 9)
+            {
+                throw new PipelineException("Sampling ratio can only be set when listening to processes running System.Diagnostics.DiagnosticSource 9 or greater");
+            }
+        }
+
+        protected override async Task OnEventSourceAvailable(EventPipeEventSource eventSource, Func<Task> stopSessionAsync, CancellationToken token)
+        {
+            await ExecuteCounterLoggerActionAsync((logger) => logger.PipelineStarted(token)).ConfigureAwait(false);
+
+            eventSource.Dynamic.All += traceEvent => {
+                try
+                {
+                    if (traceEvent.TryGetActivityPayload(out ActivityPayload activity))
+                    {
+                        foreach (IActivityLogger logger in _loggers)
+                        {
+                            try
+                            {
+                                logger.Log(
+                                    in activity.ActivityData,
+                                    activity.Tags);
+                            }
+                            catch (ObjectDisposedException)
+                            {
+                            }
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                }
+            };
+
+            using EventTaskSource<Action> sourceCompletedTaskSource = new(
+                taskComplete => taskComplete,
+                handler => eventSource.Completed += handler,
+                handler => eventSource.Completed -= handler,
+                token);
+
+            await sourceCompletedTaskSource.Task.ConfigureAwait(false);
+
+            await ExecuteCounterLoggerActionAsync((logger) => logger.PipelineStopped(token)).ConfigureAwait(false);
+        }
+
+        private async Task ExecuteCounterLoggerActionAsync(Func<IActivityLogger, Task> action)
```

Comment:
```suggestion
        private async Task ExecuteActivityLoggerActionAsync(Func<IActivityLogger, Task> action)
```
Did you intend to name this 'CounterLogger'? I'm guessing this is a copy/paste that didn't get renamed.